aspnetcore/security/key-vault-configuration.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Azure Key Vault configuration provider | Microsoft Docs
+author: guardrex
+description: How to use the Azure Key Vault Configuration Provider to configure an application using name-value pairs loaded at runtime.
+keywords: ASP.NET Core, configuration, Azure Key Vault
+ms.author: riande
+manager: wpickett
+ms.date: 01/22/2017
+ms.topic: article
+ms.assetid: 0292bdae-b3ed-4637-bd67-19b9bb8b65cb
+ms.prod: aspnet-core
+uid: security/key-vault-configuration
+---
+# Azure Key Vault configuration provider
+
+By [Luke Latham](https://github.com/GuardRex) and [Andrew Stanton-Nurse](https://github.com/anurse)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/security/key-vault-configuration/sample)
+
+This document explains how to use the [Microsoft Azure Key Vault](https://azure.microsoft.com/services/key-vault/) configuration provider to load application configuration values from Azure Key Vault. Azure Key Vault is a cloud-based service that helps you safeguard cryptographic keys and secrets used by applications and services. Common scenarios include controlling access to sensitive configuration data and meeting the requirement for FIPS 140-2 Level 2 validated Hardware Security Modules (HSM's) when storing configuration data.
+
+## Package
+To use the provider, add a reference to the [`Microsoft.Extensions.Configuration.AzureKeyVault`](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.AzureKeyVault/) package. The provider depends on .NET Framework 4.5.1 or .NET Standard 1.5 or higher. This feature is available for apps that target ASP.NET Core 1.1.0 or higher.
+
+## Application configuration
+You can explore the provider with the [sample application](https://github.com/aspnet/Docs/tree/master/aspnetcore/security/key-vault-configuration/sample). Once you've established a key vault and created a pair of secrets in the vault by [following the guidance below](#creating-key-vault-secrets-and-loading-configuration-values), the sample application will securely load the secret values into its configuration and display them in a webpage.
```

Comment:
The rule I had for myself was that in each paragraph where the word(s) appear, the first instance is "application." Each additional instance in that paragraph is "app." That's not right, I know.

> Don't be too aggressive on changing application to app. In many cases in this doc application is preferred.

> we prefer app over application where appropriate.

... those seem like opposite statements. Is there a firm rule I can follow?